src/order/conditionally_complete_lattice.lean

```diff
@@ -424,6 +424,42 @@ instance pi.conditionally_complete_lattice {ι : Type*} {α : Π i : ι, Type*}
 section conditionally_complete_linear_order
 variables [conditionally_complete_linear_order α] {s t : set α} {a b : α}
 
+lemma set.nonempty.Sup_mem (h : s.nonempty) (hs : finite s) : Sup s ∈ s :=
+begin
+  classical,
+  revert h,
+  apply finite.induction_on hs,
+  { simp },
+  rintros a t hat t_fin ih -,
+  rcases t.eq_empty_or_nonempty with rfl | ht,
+  { simp },
+  { rw cSup_insert t_fin.bdd_above ht,
+    by_cases ha : a ≤ Sup t,
+    { simp [sup_eq_right.mpr ha, ih ht] },
+    { simp only [sup_eq_left, mem_insert_iff, (not_le.mp ha).le, true_or] } }
+end
+
+lemma finset.nonempty.Sup_mem {s : finset α} (h : s.nonempty) : Sup (s : set α) ∈ s :=
+set.nonempty.Sup_mem h s.finite_to_set
+
+lemma set.nonempty.Inf_mem (h : s.nonempty) (hs : finite s) : Inf s ∈ s :=
+begin
```

Comment:
There should be a trick to get this via order_dual, right?

Reply:
There should be. But modifying the proof by hand was much quicker.

Reply:
I'll have a go tomorrow if another reviewer doesn't beat me to it.